### PR TITLE
8Movies movie app 

### PR DESCRIPTION
**Pull request by Chaseknis**
- Find external API
- Display Comments pop up with selected item's details
- Display the number of likes for each item on the Homepage
- Set up the project with Webpack and Jest
- Create feature: add new like 
- Add comments counter 
- Add all items counter on the Homepage
- Display Comments pop up with selected item's details
- Add tests for comments counter
-  Add tests for items counter 
- Run linters for errors
- Add README.md file

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const comments = document.querySelectorAll('.buttons_wrapper');
+  const likes = await getAllLikes();
+  comments.forEach((element) => {
+    element.addEventListener('click', () => {
+      const id = element.getAttribute('id');
+      popup(id, likes);
+    });
+  });
+};
+
```
Comment (ishimwezachee) [inline src/index.js:70]: - [x]  You project should contain comments that matches their relative item.  When I try to comment on a specific item, and click to the other item's comment section, I am getting previous comment to the new item as well. Try to fix it , so that every time a user try to comment, on a specific movies, it will not appear to the other movies as well. 
Added comments to this movies 
image
The other movies have the same comments 
image

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin'); // eslint-disable-line
+
+module.exports = {
```
Comment (ishimwezachee) [inline webpack.config.js:4]: - [x]  Try to adjust you pop up so that user can see everything on your card, such the name of the movies, number of comment and likes etc. It was perfect on mobile view, but on laptop I couldn't see them well. Or maybe you can consider the option of using scroll to pop, that can fix that issue as well. Apply that to make your project more better and clean.

File: src/index.js (hunk 1 above)
Comment (Chaseknis) [reply src/index.js:70]: @ishimwezachee Hi, the comments take time to load that's why at first they show up but the page gets to load the right comments after loading properly. if there’s anything that you'd suggest to make the page load faster please let us know.

File: src/modules/test/commentsCount.test.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  test('returns the number of comments', async () => {
+    const comment = [{ id: 1 }, { id: 2 }, { id: 3 }];
+    fetch.mockResolvedValue({ json: () => Promise.resolve(comment) });
+    const result = await commentCounter();
+    expect(result.length).toBe(3);
+  });
+
+  test('returns the comments object', async () => {
+    const comment = [{ id: 1 }, { id: 2 }, { id: 3 }];
+    fetch.mockResolvedValue({ json: () => Promise.resolve(comment) });
```
Comment (ishimwezachee) [inline src/modules/test/commentsCount.test.js:18]: - [x]  You are supposed to cover the tests of all your Counter module, you've tested comment and movies counter, but you didn't cover the like counter. please implement the test for like counter as it was suggested in the requirement . This will make your project more professional.

File: src/modules/test/commentsCount.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  beforeEach(() => {
+    fetch.mockReset();
+  });
+
+  test('returns the number of comments', async () => {
+    const comment = [{ id: 1 }, { id: 2 }, { id: 3 }];
+    fetch.mockResolvedValue({ json: () => Promise.resolve(comment) });
+    const result = await commentCounter();
+    expect(result.length).toBe(3);
+  });
```
Comment (ishimwezachee) [inline src/modules/test/commentsCount.test.js:14]: - [x]  In tests of counter modules you are supposed to consider all possible edge cases, for instance, the scenario in which there 0 items on page or 0 comments, you should consider those to all the modules and test them as well to make you project full the requirements. Remember writing good tests make your project more stable.

File: src/index.js (hunk 1 above)
Comment (Chaseknis) [reply src/index.js:70]: Hi @ishimwezachee we are solving it thank you.

File: src/modules/movies.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const createApp = async () => {
+  const key = localStorage.getItem('apiKey');
+  if (key === null || key === undefined || key === '') {
+    const rawResponse = await fetch('https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/', {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      // body: JSON.stringify({a: 1, b: 'Textual content'})
```
Comment (ishimwezachee) [inline src/modules/movies.js]: - [x]  You should remove un necessary comment in you project codes.  This will  make  your project  to have good clean codes.

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @Chaseknis  adn @LDouglasOT , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
image

## Highlights 

- Project is running successfully ✅
- Well  described pull request  ✅
- Linter are passing  ✅
- Git flow  ✅


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/modules/test/commentsCount.test.js (hunk 3 above)
Comment (LDouglasOT) [reply src/modules/test/commentsCount.test.js:18]: We have added the likes counter to the test cases

File: src/modules/test/commentsCount.test.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+describe('commentCounter', () => {
+  beforeEach(() => {
+    fetch.mockReset();
+  });
+
+  test('returns the number of comments', async () => {
+    const comment = [{ id: 1 }, { id: 2 }, { id: 3 }];
+    fetch.mockResolvedValue({ json: () => Promise.resolve(comment) });
+    const result = await commentCounter();
+    expect(result.length).toBe(3);
```
Comment (topeogunleye) [inline src/modules/test/commentsCount.test.js:13]: - [x] As suggested by the previous reviewer, please kindly add all edge test cases. E.G
- the scenario in which there 0 items on page or 0 comments

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  entry: {
```
Comment (topeogunleye) [inline webpack.config.js:5]: - [x] [OPTIONAL] It's be a great idea to add an overflow to auto to your modal on desktop in order to make the page fully visible to the user.

image

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @Chaseknis, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Good commit messages. :heavy_check_mark:
* No linter errors. :heavy_check_mark:
* Followed Github flow :heavy_check_mark:
* Professional README file. :heavy_check_mark:
* PR has a good title and summary :heavy_check_mark:

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/test/commentsCount.test.js (hunk 6 above)
Comment (Chaseknis) [reply src/modules/test/commentsCount.test.js:13]: Thank you @topeogunleye we are fixing it

Comment (topeogunleye) [reply src/modules/test/commentsCount.test.js:13]: Great 👍 , Keep up the great work

Comment (Chaseknis) [reply src/modules/test/commentsCount.test.js:13]: @ishimwezachee @topeogunleye Hey guys, if any of you is available, can you please check out our PR, we can’t seem to get anyone for the review.

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# STATUS: APPROVED :tada: :clap: :champagne: 
Hi @Chaseknis  and @LDouglasOT ,

You are amazing :smiley: :clap: !  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

 

## Highlights :clap: :tophat: 
- [x] All test are passing :+1: 
- [x] No linter errors :+1: 
## Optional suggestions :small_airplane: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
